megamek/src/megamek/server/Server.java

```diff
@@ -9832,26 +9820,24 @@ public int processTeleguidedMissileCFR(int playerId, List<Integer> targetIds, Li
                 } catch (InterruptedException e) {
                     return 0;
                 }
+
                 // Get the packet, if there's something to get
-                ReceivedPacket rp;
-                if (cfrPacketQueue.size() > 0) {
-                    rp = cfrPacketQueue.poll();
-                    int cfrType = rp.packet.getIntValue(0);
-                    // Make sure we got the right type of response
-                    if (cfrType != Packet.COMMAND_CFR_TELEGUIDED_TARGET) {
-                        MegaMek.getLogger().error("Expected a " 
-                                + "COMMAND_CFR_TELEGUIDED_TARGET CFR packet, " + "received: " + cfrType);
-                        continue;
-                    }
-                    // Check packet came from right ID
-                    if (rp.connId != playerId) {
-                        MegaMek.getLogger().error("Expected a " 
-                                + "COMMAND_CFR_TELEGUIDED_TARGET CFR packet " + "from player  " + playerId
-                                + " but instead it came from player " + rp.connId);
-                        continue;
-                    }
-                    return (int)rp.packet.getData()[1];
-                } // If no packets, wait again
+                ReceivedPacket rp = cfrPacketQueue.poll();
+                int cfrType = rp.packet.getIntValue(0);
```

Comment:
Won't this lead to an exception when the queue is empty in contrast with the old implementation?

Reply:
Nope... we already have an isEmpty check just a few lines before, and immediately before this assignment.
```
try {
    while (cfrPacketQueue.isEmpty()) {
        cfrPacketQueue.wait();
    }
} catch (InterruptedException e) {
    return false;
}

ReceivedPacket rp;
if (cfrPacketQueue.size() > 0) {
    rp = cfrPacketQueue.poll();
```

Reply:
Ah ok didn't see that in the diff view.